Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -279,51 +265,24 @@ namespace OpenInfraPlatform {
 					} // end if IfcBSplineCurve
 
 					// (3/6) IfcCompositeCurve SUBTYPE OF IfcBoundedCurve
-					std::shared_ptr<typename IfcEntityTypesT::IfcCompositeCurve> composite_curve =
-						std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcCompositeCurve>(bounded_curve.lock());
-					if (composite_curve) {
-
-						std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcCompositeCurveSegment> > segments;
-						segments.resize(composite_curve->Segments.size());
-
-						std::transform(
-							composite_curve->Segments.begin(),
-							composite_curve->Segments.end(),
-							segments.begin(),
-							[](auto &it) { return it.lock(); });
-
-						std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcCompositeCurveSegment> >::iterator it_segments =
-							segments.begin();
-
-						for (; it_segments != segments.end(); ++it_segments) {
-							std::shared_ptr<typename IfcEntityTypesT::IfcCompositeCurveSegment> segment = (*it_segments);
-							std::shared_ptr<typename IfcEntityTypesT::IfcCurve> segment_curve = segment->ParentCurve.lock();
-
-							std::vector<carve::geom::vector<3>> segment_vec;
-							convertIfcCurve(segment_curve, segment_vec, segmentStartPoints);
-							if (!segment_vec.empty()) {
-								GeomUtils::appendPointsToCurve(segment_vec, targetVec);
-							}
-						}
-						return;
+					else if (bounded_curve.isOfType<typename IfcEntityTypesT::IfcCompositeCurve>()) {
+						return convertIfcCompositeCurve(bounded_curve.as<typename IfcEntityTypesT::IfcCompositeCurve>(), targetVec, segmentStartPoints);

```

Comment:
triming parameters need to be passed on, similarly to `convertIfcIndexedPolyCurve` - even if they are ignored at the moment.

Reply:
Apply also to other functions called in these `if else` statements.

Reply:
Resolved: b1f9677